src/test/java/teammates/common/datatransfer/attributes/NotificationAttributesTest.java

```diff
@@ -0,0 +1,270 @@
+package teammates.common.datatransfer.attributes;
+
+import java.time.Instant;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.NotificationType;
+import teammates.storage.entity.Notification;
+import teammates.test.BaseTestCase;
+
+/**
+ * SUT: {@link NotificationAttributes}.
+ */
+public class NotificationAttributesTest extends BaseTestCase {
+    private static final Instant STARTTIME_ONE = Instant.now().plusSeconds(3600);
+    private static final Instant ENDTIME_ONE = Instant.now().plusSeconds(7200);
+    private static final Instant STARTTIME_TWO = Instant.now().plusSeconds(1000);
+    private static final Instant ENDTIME_TWO = Instant.now().plusSeconds(10000);
+
+    @Test
+    public void testValueOf_withAllFieldPopulatedNotificationAttributes_shouldGenerateAttributesCorrectly() {
+        Notification notification = new Notification("valid-notification-id",
+                STARTTIME_ONE, ENDTIME_ONE,
+                NotificationType.DEPRECATION, NotificationTargetUser.INSTRUCTOR,
+                "valid notification title", "valid notification message", false,
+                Instant.now(), Instant.now());
```

Comment:
If these data are going to be used frequently across different tests, you may want to use a helper method like `generateTypicalNotificationAttributesObject` to generate a `NotificationAttributes` object for you and reuse them across tests.

Reply:
The builder might have slight variations in certain fields and in numbers of fields. If we abstract a method out for that, my concern is it will become just a wrapper around the builder, which might not be necessary.

Reply:
Yes, for cases where we need to test with different values we can build the `NotificationAttributes` separately, but for typical `NotificationAttributes` we can use a helper method method like `generateTypicalNotificationAttributesObject` to improve readability.

Reply:
Thank you for the suggestion! I have changed accordingly in the new commit.